lib/scholar/model_selection.ex

```diff
@@ -105,9 +106,129 @@ defmodule Scholar.ModelSelection do
         ]
       >
   """
-  def cross_validate(x, y, folding_fun, scoring_fun) do
+  def cross_validate(x, y, folding_fun, scoring_fun, opts \\ []) do
```

Comment:
Let's also add guards here:

```suggestion
  def cross_validate(x, y, folding_fun, scoring_fun, opts \\ []) when is_function(folding_fun, 1) and is_list(opts) do
```